clang/test/CIR/Lowering/struct.c

```diff
@@ -0,0 +1,89 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -fclangir -emit-llvm %s -o %t.ll -fclangir-call-conv-lowering
+// RUN: FileCheck --input-file=%t.ll %s -check-prefix=LLVM
+
+typedef struct {
+  int a, b;
+} S;
+
+// LLVM: @init(i64 %[[#V0:]])
```

Comment:
This testcase is great, thanks. The test should also show pre-lowering and post-lowering in CIR terms, see other tests from `clang/test/CIR/Transforms/Target/x86_64/`, most important it should show the casts.

Reply:
Noted. I will update these next week.

Reply:
I would advise against these tests.

They are too monolithic as they test multiple areas of the pipeline and are not quite related to the changes in this PR. These usually are extra work to maintain.

The CC lowering test already covers what this PR changes, what comes after is another pass' responsibility.

Reply:
I'm talking about the location, I think this is the best subdir to use for callconv tests, @sitio-couto do you have an alternative suggestion?

> The CC lowering test already covers what this PR changes, what comes after is another pass' responsibility.

It's totally fine if new tests in this subdir test for LLVM output

Reply:
@bcardosolopes I see what you mean about the location, and I agree that from the options we have, your suggestion is the best.

That said, I’m not in favor of adding the tests at all. IMO, they introduce more upkeep than benefits for the reasons mentioned. The CC lowering tests already feel a bit too monolithic. Ideally, they should focus on CIR-to-CIR testing, while CIR-to-LLVM is handled by lowering tests.

Reply:
I agree with @sitio-couto that we should focus in CIR-to-CIR testing under transform tests, amd LLVM in lowering tests. For now, I will add the CIR equivalent under the transform tests, and the LLVM equivalent under the lowering tests. Let me know what you think.

Reply:
> Ideally, they should focus on CIR-to-CIR testing, while CIR-to-LLVM is handled by lowering tests. C/C++ to LLVM IR is another suite of tests of its own, I suppose.

In an ideal world that would be right, but we need AST for some of the pieces and we don't yet have a mechanism to serialize them, so CIR to CIR is only applicable to a handful of cases. Note that `test/clang/CIR/Lowering` is already a mix of testing both LLVM and CIR in the same file AND standalone CIR tests. We should follow this pattern until we have a AST serialization story that will break the monolithic behavior for good.

> For now, I will add the CIR equivalent under the transform tests, and leave the LLVM equivalent under the lowering tests. Let me know what you think.

I just moved things around and we now have `test/clang/CIR/CallConvLowering/AArch64`, please put `struct.c` there and test both CIR and LLVM output in the same file (note that this is a very common pattern in Lowering subdir).

Reply:
>  we need AST for some of the pieces [...] so CIR to CIR is only applicable to a handful of cases.

Agreed, and in this case, it applies: the test doesn't rely on AST info. While these tests are necessary, we should minimize them when possible, as you pointed out, it's the "ideal."

> `test/clang/CIR/Lowering` is already a mix of testing both LLVM and CIR in the same file AND standalone CIR tests. We should follow this pattern until we have a AST serialization story that will break the monolithic behavior for good.

I think we should follow said pattern when it makes sense; otherwise, we'll introduce unnecessary redundancy. Lowering tests are tightly coupled with LLVM Dialect and IR, requiring both `cir-opt` and `cir-translate`, so it's coherent to include both. This approach is also common in CodeGen, with the goal of ensuring CIRGen emits the same LLVM IR as CodeGen, AFAIK, which also seems coherent.

In any case, I fine with your call.